### PR TITLE
Made changes to disable save button in appropriate times (logged in m…

### PR DESCRIPTION
**Pull request by RaiBishal**
### Description of work
Feedback from the IBEX server is now reflected on the GUI.

### Ticket

https://github.com/ISISComputingGroup/IBEX/issues/4796

### Acceptance criteria

- [x] The "save" button in the configuration/component editing dialogs is greyed out if I would not be allowed to save. A sensible warning is displayed to the user if saving is disabled

- [x] I can use "save as" to save any configuration with a different name, regardless of whether the original configuration was protected or not.

- [x] I am not allowed to use "save as" to overwrite an existing, protected configuration unless I am in manager mode. This should give an appropriate error to the user, e.g. via a pop-up dialog box, but allowing the user to try again with a different name.

- [x] If I "save as" a protected configuration and I am not in manager mode, the copy does not have the protected flag set. A warning needs to displayed to the user at an appropriate time if the configuration is being "downgraded" in this way. If I am in manager mode, the protected flag remains set.

- [x] Protected configs show a warning on the delete dialogue and can not be deleted if not in manager mode.

- [x] If I am in the middle of editing a config/component and manager mode changes while I am editing it, the button should become greyed/ungreyed and the warning should be displayed/hidden as appropriate. This must be dynamic (it should not just remember the state of manager mode when editing was started).

### Unit tests

*Give an overview of unit tests you have added or modified, if applicable. The aim is provide information to help the reviewer*

### System tests

*Mention any automated tests or manual tests that you have added or modified, if applicable. The aim is provide information to help the reviewer*

### Documentation
*Highlight and provide a link to any additions or changes to the documentation, if applicable. The aim is provide information to help the reviewer*

---

#### Code Review

- [x] Is the code of an acceptable quality?
- [x] Do the changes function as described and is it robust?
- [x] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?

### Final Steps
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/DeleteConfigsHandler.java [hunk 1]
```diff
@@ -48,11 +54,49 @@ public DeleteConfigsHandler() {
 	public void safeExecute(Shell shell) {
```
Comment (JamesKingWork) [inline base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/DeleteConfigsHandler.java]: I may be misunderstanding this function. Are we meant to be actually deleting the configs or simply removing them from the recently loaded list?

File: base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/DeleteConfigsHandler.java [hunk 2]
```diff
@@ -48,11 +54,49 @@ public DeleteConfigsHandler() {
...
+						MessageDialog.ERROR, new String[] {"OK"},0);
+				error.open();
+			}
+		    
 		}
 	}
+	
+	/**
+	 * opens and error dialog if there is an error
+	 * @param doNotDisplayError if dialog needs to be created or not
```
Comment (JamesKingWork) [inline base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/DeleteConfigsHandler.java]: doNotDisplayError param is not present in the method signature

File: base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/dialogs/SaveConfigDialog.java [hunk 3]
```diff
@@ -496,6 +510,19 @@ private String getErrorMessage(String name, String description) {
         if (nameMatchesCurrentConfig(name)) {
             return "Name matches the current configuration";
         }
+        
+        try {
+			if (compInProtectionMode(name) && !model.isInManagerMode()) {
+				return "Component with the name already exisist, try again in Manager mode to overwrite it";
```
Comment (JamesKingWork) [inline base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/dialogs/SaveConfigDialog.java]: Typo "exisist"

File: base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/dialogs/SaveConfigDialog.java [hunk 4]
```diff
@@ -496,6 +510,19 @@ private String getErrorMessage(String name, String description) {
...
+			}
+			if (configInProtectionMode(name) && !model.isInManagerMode()) {
+	        	return "Configuration with the name already exsist, try again in Manager mode to overwrite it";
```
Comment (JamesKingWork) [inline base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/dialogs/SaveConfigDialog.java]: Typo "exsist"

File: base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/editing/summary/SummaryPanel.java [hunk 5]
```diff
@@ -109,23 +111,30 @@ public SummaryPanel(Composite parent, int style, MessageDisplayer dialog) {
...
         lblDateModified.setLayoutData(new GridData(SWT.RIGHT, SWT.CENTER, false, false, 1, 1));
         lblDateModified.setText("Date Modified:");
 
         lblDateModifiedField = new Label(cmpSummary, SWT.NONE);
+        
+        warning = new Label(cmpSummary, SWT.NONE);
+        warning.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, false, false, 2, 1));
+        warning.setText("Info : To modify/save a protected configuration/component you have to be in Manager Mode");
+        warning.setVisible(false);
+        warning.setForeground(new Color(getDisplay(),252, 169, 3));
```
Comment (JamesKingWork) [inline base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/editing/summary/SummaryPanel.java]: Orange on grey text is not easy to read

File: base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/DeleteConfigsHandler.java (hunk 1 above)
Comment (RaiBishal) [reply base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/DeleteConfigsHandler.java]: Function description is vague, it opens the delete configs and also deletes the config with `configService.uncheckedWrite((dialog.selectedConfigs()))` it then removes deleted configs from recently used list using `Configurations.getInstance().removeNameFromRecentlyLoadedConfigList(item)` if no error is detected.

Comment (RaiBishal) [reply base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/DeleteConfigsHandler.java]: Function description is vague, it opens the delete configs and also deletes the config with configService.uncheckedWrite((dialog.selectedConfigs())) it then removes deleted configs from recently used list using Configurations.getInstance().removeNameFromRecentlyLoadedConfigList(item) if no error is detected.

File: base/uk.ac.stfc.isis.ibex.configserver/src/uk/ac/stfc/isis/ibex/configserver/editing/EditableConfiguration.java [hunk 6]
```diff
@@ -152,7 +165,10 @@ public EditableConfiguration(
 		this.synoptic = config.synoptic();
 		this.isProtected = config.isProtected();
 		this.allIocs = new ArrayList<>();
-		
+		this.managerMode = ManagerModeModel.getInstance();
+		this.enableOrDisableSaveButton = true;
```
[the remark this answers is not in the document]
Comment (JamesKingWork) [reply base/uk.ac.stfc.isis.ibex.configserver/src/uk/ac/stfc/isis/ibex/configserver/editing/EditableConfiguration.java]: Sorted.

File: base/uk.ac.stfc.isis.ibex.configserver/src/uk/ac/stfc/isis/ibex/configserver/editing/EditableConfiguration.java [hunk 7]
```diff
@@ -29,6 +29,8 @@
 import java.util.Map;
 import java.util.NoSuchElementException;
 
+import org.eclipse.jface.dialogs.MessageDialog;
```
[the remark this answers is not in the document]
Comment (JamesKingWork) [reply base/uk.ac.stfc.isis.ibex.configserver/src/uk/ac/stfc/isis/ibex/configserver/editing/EditableConfiguration.java]: Sorted.

File: base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/DeleteComponentsHandler.java [hunk 8]
```diff
@@ -56,15 +60,32 @@ public void safeExecute(Shell shell) {
...
             } else {
                 displayWarning(selectedDependencies, shell);
-                safeExecute(shell); // Re-open selection dialog.
+                safeExecute(shell);
+            }
+            try {
+                if (!ManagerModeModel.getInstance().isInManagerMode()) {
+                    for (String item: dialog.selectedConfigs()) {
+                        if (compNamesWithFlags.get(item)) {
+                            dispalyError(shell);
```
[the remark this answers is not in the document]
Comment (JamesKingWork) [reply base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/DeleteComponentsHandler.java]: Sorted.

File: base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/DeleteConfigsHandler.java (hunk 1 above)
Comment (JamesKingWork) [reply base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/DeleteConfigsHandler.java]: This makes more sense thank you

File: base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/DeleteConfigsHandler.java (hunk 2 above)
Comment (JamesKingWork) [reply base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/DeleteConfigsHandler.java]: Sorted.

File: base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/dialogs/SaveConfigDialog.java (hunk 3 above)
Comment (JamesKingWork) [reply base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/dialogs/SaveConfigDialog.java]: Sorted.

File: base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/dialogs/SaveConfigDialog.java (hunk 4 above)
Comment (JamesKingWork) [reply base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/dialogs/SaveConfigDialog.java]: Sorted.

**Pull request merged**